ipa-core/src/utils/arraychunks.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{ff::Fp61BitPrime, secret_sharing::SharedValue};
+
+#[allow(dead_code)]
+pub trait ArrayChunkIterator: Iterator {
```

Comment:
Is it worth implementing ExactSizeIterator when the input is also an ExactSizeIterator?

Reply:
I think right now, I only have one use-case for this for which I wouldn't need it. In our use case it is also correct to fill the array with zeros. In a general use case it might be a bit weird, so I didn't want to make it more general.

Reply:
I think that this should implement ExactSizeIterator where possible, if only because it makes some of the code might like to depend on it (for accurate pre-allocation, if nothing else).